.github/workflows/update-dependencies.yaml

```diff
@@ -0,0 +1,115 @@
+name: update-dependencies
+on:
+  workflow_call:
+    inputs:
+      component:
+        type: string
+        required: true
+      write-lock:
+        type: string
+        required: true
+      label:
+        type: string
+        required: true
+
+jobs:
+  init:
+    runs-on: [ "self-hosted", "builder" ]
+    outputs:
+      metadata: ${{ steps.init.outputs.metadata }}
+      parameters: ${{ steps.init.outputs.parameters }}
+    steps:
+      - id: init
+        name: Init Workflow
+        uses: hawk-ai-aml/github-actions/workflow-init@master
+        with:
+          component: ${{ inputs.component }}
+          profile: hawk
+          update-kustomize: false
+          
+  lock-dependencies:
+    needs: [ init ]
+    runs-on: [ "self-hosted", "builder" ]
+    outputs:
+      result: ${{ steps.commit-and-push.outputs.result }}
+    steps:
+      - id: init
+        name: Init Job
+        uses: hawk-ai-aml/github-actions/job-init@master
+
+      - name: Set up JDK
+        uses: actions/setup-java@v3
+        with:
+          java-version: ${{ fromJson(needs.init.outputs.parameters).java-version }}
+          distribution: temurin
+
+      - uses: actions/checkout@v3
+        with:
+          token: ${{ secrets.REPO_ACCESS_PAT }}
+          ref: update-dependencies
+
+      - name: Fetch and reset update-dependencies to master
+        run: |
+          git fetch origin master
+          git reset --hard origin/master
```

Comment:
Why do we need to reset the master?

Reply:
Because, I want to have update-dependencies branch be ahead of master with only one commit containing lockfiles updates
1. If I did rebase, then in case of second scheduled run and further update we would have two commits in update-dependencies
2. If I did nothing and someone merged a feature branch into master, we'd have potential conflicts